python/ray/worker.py

```diff
@@ -1222,23 +1222,26 @@ def connect(node,
         serialized_job_config, node.metrics_agent_port)
     worker.gcs_client = worker.core_worker.get_gcs_client()
 
-    # Notify raylet that the core worker is ready.
-    worker.core_worker.notify_raylet()
-
     # Create an object for interfacing with the global state.
     # Note, global state should be intialized after `CoreWorker`, because it
     # will use glog, which is intialized in `CoreWorker`.
     ray.state.state._initialize_global_state(
         node.redis_address, redis_password=node.redis_password)
-    if mode == SCRIPT_MODE:
+    # If it's a driver and it's not coming from ray client, we'll prepare the
+    # environment here. If it's ray client, the environmen will be prepared
+    # at the server side.
+    if mode == SCRIPT_MODE and not job_config.client_job:
         runtime_env.upload_runtime_env_package_if_needed(job_config)
     elif mode == WORKER_MODE:
         # TODO(ekl) get rid of the env var hack and get runtime env from the
         # task spec and/or job config only.
         job_config = os.environ.get("RAY_RUNTIME_ENV_FILES")
         job_config = [job_config] if job_config else \
             worker.core_worker.get_job_config().runtime_env.uris
-        runtime_env.ensure_runtime_env_setup(job_config)
+        assert len(runtime_env.ensure_runtime_env_setup(job_config)) == 0
```

Comment:
Why change it to an assert? The error message will not be as good as letting the original exception bubble.